game/plugin/skills/woodcutting/src/woodcutting.plugin.kts

```diff
@@ -0,0 +1,117 @@
+import org.apollo.cache.def.ItemDefinition
+import org.apollo.game.action.AsyncDistancedAction
+import org.apollo.game.message.impl.ObjectActionMessage
+import org.apollo.game.model.Position
+import org.apollo.game.model.World
+import org.apollo.game.model.entity.Player
+import org.apollo.game.model.entity.Skill
+import org.apollo.game.model.entity.obj.GameObject
+import org.apollo.game.plugin.skills.woodcutting.*
+import org.apollo.game.plugins.api.Definitions
+import org.apollo.game.plugins.api.woodcutting
+import org.apollo.game.plugins.api.skills
+import org.apollo.game.scheduling.ScheduledTask
+import java.util.*
+
+class WoodcuttingTarget(val objectId: Int, val position: Position, val wood: Wood) {
+
+    fun getObject(world: World): Optional<GameObject> {
+        val region = world.regionRepository.fromPosition(position)
+        val obj = region.findObject(position, objectId)
+
+        return obj
+    }
+
+    fun isSuccessful(mob: Player): Boolean {
+        return rand(100) <= wood.chance * 100
+    }
+}
+
+class WoodcuttingAction(val player: Player, val tool: Axe, val target: WoodcuttingTarget) : AsyncDistancedAction<Player>(PULSES, true, player, target.position, TREE_SIZE) {
+
+    companion object {
+        private val PULSES = 0
+        private val TREE_SIZE = 1;
+
+        fun axeFor(player: Player): Axe? {
+            return AXES
+                    .filter { it.level <= player.skills.woodcutting.currentLevel }
+                    .filter { player.equipment.contains(it.id) || player.inventory.contains(it.id) }
+                    .sortedByDescending { it.level }
+                    .firstOrNull()
+        }
+
+        /**
+         * Starts a [WoodcuttingAction] for the specified [Player], terminating the [Message] that triggered it.
+         */
+        fun start(message: ObjectActionMessage, player: Player, wood: Wood) {
+            val axe = axeFor(player)
+            if (axe != null) {
+                val action = WoodcuttingAction(player, axe, WoodcuttingTarget(message.id, message.position, wood))
+                System.out.println("Starting action")
```

Comment:
Can you remove the print statements please (and btw in kotlin this could just be `println(msg_here)`)